ddtrace/tracer/span.go

```diff
@@ -319,13 +332,94 @@ func (s *span) finish(finishTime int64) {
 	}
 	s.finished = true
 
+	if t, ok := internal.GetGlobalTracer().(*tracer); ok {
+		// we have an active tracer
+		feats := t.features.Load()
+		if feats.Stats && shouldComputeStats(s) {
+			// the agent supports computed stats
+			select {
+			case t.stats.In <- newAggregableSpan(s, t.config.version):
+				// ok
+			default:
+				log.Error("Stats channel full, disregarding span.")
+			}
+		}
+		if feats.DropP0s {
+			// the agent supports dropping p0's in the client
+			if shouldDrop(s) {
+				// ...and this span can be dropped
+				atomic.AddUint64(&t.droppedP0Spans, 1)
+				if s == s.context.trace.root {
+					atomic.AddUint64(&t.droppedP0Traces, 1)
+				}
+				return
+			}
+		}
+	}
 	if s.context.drop {
 		// not sampled by local sampler
 		return
 	}
 	s.context.finish()
 }
 
+// newAggregableSpan creates a new summary for the span s, within an application
+// version version.
+func newAggregableSpan(s *span, version string) *aggregableSpan {
+	var statusCode uint32
+	if sc, ok := s.Meta["http.status_code"]; ok && sc != "" {
+		if c, err := strconv.Atoi(sc); err == nil {
+			statusCode = uint32(c)
+		}
+	}
+	key := aggregation{
+		Name:       s.Name,
+		Resource:   s.Resource,
+		Service:    s.Service,
+		Type:       s.Type,
+		Hostname:   s.Meta[keyHostname],
```

Comment:
env, hostname, version are global tracer configs, you don't need to add them here and can reduce the cost of map hash
They should only be set if enabled by config:
- [hostname](https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/option.go#L123-L132)
- [env](https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/option.go#L133-L135)
- [version](https://github.com/DataDog/dd-trace-go/blob/v1/ddtrace/tracer/option.go#L140-L142)

Reply:
I guess you're right. Perhaps it's bad design that they are set as tags too. Changed it.